solutions/baseline/variables.tf

```diff
@@ -0,0 +1,286 @@
+  default     = true
+}
+
+variable "addons" {
+  type = object({
+    debug-tool                = optional(string)
+    image-key-synchronizer    = optional(string)
+    openshift-data-foundation = optional(string)
+    vpc-file-csi-driver       = optional(string)
+    static-route              = optional(string)
+    cluster-autoscaler        = optional(string)
+    vpc-block-csi-driver      = optional(string)
+    ibm-storage-operator      = optional(string)
+  })
+  description = "Map of OCP cluster add-on versions to install (NOTE: The 'vpc-block-csi-driver' add-on is installed by default for VPC clusters and 'ibm-storage-operator' is installed by default in OCP 4.15 and later, however you can explicitly specify it here if you wish to choose a later version than the default one). For full list of all supported add-ons and versions, see https://cloud.ibm.com/docs/containers?topic=containers-supported-cluster-addon-versions"
+  nullable    = false
+  default     = {}
+}
+
+variable "manage_all_addons" {
+  type        = bool
+  default     = false
+  nullable    = false # null values are set to default value
+  description = "Instructs Terraform to manage all cluster addons, even if addons were installed outside of the module. If set to 'true' this module will destroy any addons that were installed by other sources."
+}
+
+variable "cluster_config_endpoint_type" {
+  description = "Specify which type of endpoint to use for for cluster config access: 'default', 'private', 'vpe', 'link'. 'default' value will use the default endpoint of the cluster."
+  type        = string
+  default     = "default"
+  nullable    = false # use default if null is passed in
+}
+
+variable "enable_ocp_console" {
+  description = "Flag to specify whether to enable or disable the OpenShift console."
+  type        = bool
+  default     = true
+}
+
+variable "kms_config" {
+  type = object({
+    crk_id           = string
+    instance_id      = string
+    private_endpoint = optional(bool, true) # defaults to true
+    account_id       = optional(string)     # To attach KMS instance from another account
+    wait_for_apply   = optional(bool, true) # defaults to true so terraform will wait until the KMS is applied to the master, ready and deployed
+  })
+  description = "Use to attach a KMS instance to the cluster. If account_id is not provided, defaults to the account in use."
+  default     = null
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module."
+  default     = []
+}
+
+variable "vpc_id" {
+  type        = string
+  description = "Id of the VPC instance where this cluster will be provisioned"
+}
+
+variable "existing_cos_id" {
```

Comment:
To ensure consistency with other DAs, this should be renamed `existing_cos_instance_crn`